kythe/docs/rfc/3339.md

```diff
@@ -0,0 +1,348 @@
+    
+    The reply includes the input file(s) on which to begin analysis,
+    as well as the output file to which to emit analysis output. It
+    reflects some of the kythe.proto.CompilationUnit structure. The
+    driver (and driver environment) MUST make contents available for
+    each file referenced in the compilation record at the file system
+    path specified in the compilation record.
+
+    The output file should contain the content appropriate to the
+    analysis-type, encoded as specified in the `InitRequest`. For
+    Kythe index analyses, this would be `kythe.common.Entry`
+    objects. Other analyziers might emit a "findings" structure.
+
+    Note that contents need not be literally copied. An overlay
+    filesystem, hardlinks, symlinks, or other may be used.
+
+    Example:
+
+        » {"jsonrpc":"2.0","id":2,"method":"analyze","params":{"types":["kythe/indexer/go"]}}
+        « {"jsonrpc":"2.0","id":2","result":{"id":101,"input":"path/to/main.go"}}
+
+-   `vname(VNameRequest)`
+
+    Requests the complete VName to emit for the specified path and signature.
+
+-   `log(LogRequest)`
+
+    A notification from the analyzer to the environment, containing a message
+    string to be logged in association with the specified analysis ID. The
+    environment does not reply.
+
+    Example:
+
+        » {"jsonrpc":"2.0","method":"log","params":{"id":101,"message":{...}}}
+
+-   `done(LogRequest)`
+
+    A notification from the analyzer to the environment, signifying that the
+    analyzer has completed work on the specified analysis ID. The environment
+    does not reply. After receiving this notification, the driver MUST discard
+    any further requests for the analysis ID (and return an error, if
+    appropriate) unless and until it issues a new analysis reusing that ID.
+
+    Example:
+
+        » {"jsonrpc":"2.0","method":"done","params":{"id":101}}
+
+### Errors
+
+In addition to the standard errors defined by JSON-RPC, this protocol defines
+these errors:
+
+-   `ProtocolError` (code: -1). Indicates a violation of the protocol.
+-   `FileNotFound` (code: -2). Indicates that a requested file was not found.
+
+## Example Transcript
+
+The following is an example transcript of an interaction between an analyzer and
+its environment. The marker "»" denotes a request sent by the analyzer to the
+environment; the marker "«" denotes a reply sent by the environment to the
+analyzer. Lines prefixed with `--` are comments.
+
+```
+-- Protocol handshake.
+
+» {"jsonrpc":"2.0", "id":1, "method":"init", "params":{"protocol":"kythe1","output-encoding":"protobuf"}}
+« {"jsonrpc":"2.0", "id":1, "result":{"protocol":"kythe1"}}
+
+-- Request two analysis tasks. The analyzer may process as many analysis
+-- tasks concurrently as it wishes.
+
+» {"jsonrpc":"2.0", "id":2, "method":"analysis", "params":{"types":["kythe/indexer/go"]}}
+« {"jsonrpc":"2.0", "id":2, "result":{"id":101, "input":"path/to/file1.go"}}
```

Comment:
s/input/inputs/ (it's specified as a list)

this also doesn't include the output file path, which seems like it would be required to do anything meaningful?